pages/speakers/[speaker].jsx

```diff
@@ -0,0 +1,84 @@
+import groq from "groq";
+import classNames from "classnames/bind";
+import styles from "./speakers.module.scss";
+import client from "../../modules/sanity/client";
+
+const cx = classNames.bind(styles);
+
+export default function Speaker({ person }) {
+  const { firstName, lastName, professionalTitle, bio } = person;
+  console.log("speaker in component", person);
+  return (
+    <section className={cx("speaker-container")}>
+      <div className={cx("image-placeholder")}></div>
+      <p className={cx("speaker-name")}>
+        {firstName} {lastName}{" "}
+      </p>
+    </section>
+  );
+}
+
+let fName;
+let lName;
+
+const query = groq`*[_type == "person"] {
+  firstName,
+  lastName
+}`;
+
+// Sets all the possible paths for our speakers
+export async function getStaticPaths() {
+  try {
+    const people = await client.fetch(query);
+    console.log("PEOPLE", people);
+    const paths = people.map((person) => ({
+      params: { speaker: person.firstName + "-" + person.lastName },
```

Comment:
This works but is brittle. If there's no `firstName` or `lastName` properties this will throw a runtime error. 
Doing `person?.firstName` avoids the runtime error but you lose visibility into when errors might be happening - it fails silently. 
I think a reusable name function that handles full name logic could be reused in a lot of places so perhaps is a good thing to look for in npm or create ourselves.